linker.go

```diff
@@ -292,24 +295,45 @@ fixups:
 	for {
 		ins := iter.Ins
 
-		if !ins.IsKfuncCall() {
+		metadata := ins.Metadata.Get(kfuncMetaKey{})
+		if metadata == nil {
 			if !iter.Next() {
 				// break loop if this was the last instruction in the stream.
 				break
 			}
 			continue
 		}
 
+		if !ins.IsKfuncCall() && !ins.OpCode.IsDWordLoad() {
```

Comment:
Nit: can you fold this into the `if ins.IsKfuncCall()` check + fixup below? Seems more natural there and we don't duplicate the conditions.